apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -322,10 +355,49 @@ def mlset(self, keys, value):
         container, key = find_ml(self._data, keys)
         container[key] = value
 
+    def mlset_cast(self, keys, value):
+        type_msg = "The type of {} should be {}."
+
+        if isinstance(keys, str):
+            key_list = keys.split(".")
+        elif not isinstance(keys, list):
+            key_list = list(keys)
+
+        schema_keys = list(chain.from_iterable(zip(['properties'] * len(key_list), key_list)))
+
+        container, key = find_ml(self.validator.schema, schema_keys)
+
+        if key in container:
+            container = container[key]
+            if "type" in container:
+                val_type = container["type"]
+                if val_type == "integer":
+                    if value.isdigit():
+                        value = int(value)
+                    else:
+                        raise TypeError(type_msg.format(keys, val_type))
+                elif val_type == "boolean":
+                    if value.lower == "true" or value == "1":
+                        value = True
+                    elif value.lower == "false" or value == "0":
+                        value = False
+                    else:
+                        raise TypeError(type_msg.format(keys, val_type))
+
+        self.mlset(keys, value)
+
     def mlsetwork(self, keys, value):
         container, key = find_ml(self._data, keys)
         return container.setwork(key, value)
 
     def mlsetdefault(self, keys, value):
         container, key = find_ml(self._data, keys)
         return container.setdefault(key, value)
+
+    def mldel(self, keys):
+        container, key = find_ml(self._data, keys)
+        del container[key]
+
+    def recursive_update(self, new_data):
+        update(self._data, new_data)
```

Comment:
If you say `update(self, new_data)` then both _data and _work are updated and dirty bit is set correctly.